views/moderation.js

```diff
@@ -0,0 +1,169 @@
+var mauth = require('materialized-group-auth')
+var sub = require('subleveldown')
+var EventEmitter = require('events').EventEmitter
+var { Readable, Transform } = require('readable-stream')
+var once = require('once')
+var pump = require('pump')
+var readonly = require('read-only-stream')
+
+module.exports = function (cabal, modKey, db) {
+  var events = new EventEmitter()
+  var auth = mauth(db)
+  var queue = []
+  var localKey = null
+  cabal.getLocalKey(function (err, key) {
+    if (err) return events.emit('error', err)
+    var batch = []
+    var pending = 2
+    auth.isMember({ group: '@', id: key }, function (err, m) {
```

Comment:
Won't this make *every* user joining a cabal -- who hasn't yet synced down any remote moderation messages yet -- try to publish themselves as admin? This should be OK in the sense that it won't break anything, but it does mean lots of redundant self-adminning messages.

Reply:
The admin publish is not written to a log so it doesn't leave the device.

Reply:
Right!